packages/server-core/src/user/login/login.class.ts

```diff
@@ -77,9 +79,62 @@ export class LoginService implements ServiceInterface {
       }
       if (new Date() > new Date(result.data[0].expiresAt)) {
         logger.info('Login Token has expired')
+        await this.app.service(loginTokenPath).remove(result.data[0].id)
         return { error: 'Login link has expired' }
       }
       const identityProvider = await this.app.service(identityProviderPath).get(result.data[0].identityProviderId)
+      let addToLogin = 0
+      if (result.data[0].associateUserId && params!.query?.associate === 'true') {
+        await this.app.service(identityProviderPath).patch(identityProvider.id, {
+          userId: result.data[0].associateUserId
+        })
+        await this.app.service(userLoginPath).create({
+          userId: result.data[0].associateUserId as UserID,
+          userAgent: params!.headers!['user-agent'],
+          identityProviderId: identityProvider.id,
+          ipAddress: params!.forwarded?.ip || ''
+        })
+      }
+      if (params!.query?.associate != null) addToLogin++
+      const logins = await this.app.service(userLoginPath).find({
+        query: {
+          userId: identityProvider.userId
+        }
+      })
+      //Email identity-providers are created as type email, so new vs. existing logins can't be discerned by
+      //whether the current identity-provider is a guest. We're using logins + addToLogin being 0 as a proxy for
+      //a brand-new login with that email, which should trigger the auto-association, and any total greater than
+      //0 will be seen as an email that is established and shouldn't have the auto-association trigger
+      if (identityProvider.type === 'email' && logins.total + addToLogin === 0) {
```

Comment:
this is a little indirectional using a number for essentially a boolean operation, can we make `addToLogin` a boolean and the logic here can be `logins.total === 0 && !addToLogin` and `addToLogin` can be equal to `params!.query?.associate != null`

Reply:
Fixed